src/components/look-controls.js

```diff
@@ -55,136 +68,177 @@ module.exports.Component = registerComponent('look-controls', {
     canvasEl.removeEventListener('touchend', this.onTouchEnd);
   },
 
+  /**
+   * Update orientation for mobile, mouse look, and headset cases.
+   * Mouse-drag only enabled if HMD is not active.
+   */
   updateOrientation: (function () {
     var hmdEuler = new THREE.Euler();
     return function () {
       var currentRotation;
       var deltaRotation;
+      var el = this.el;
+      var hmdQuaternion;
       var pitchObject = this.pitchObject;
-      var yawObject = this.yawObject;
-      var hmdQuaternion = this.calculateHMDQuaternion();
-      var sceneEl = this.el.sceneEl;
       var rotation;
+      var sceneEl = el.sceneEl;
+      var yawObject = this.yawObject;
+
+      hmdQuaternion = this.calculateHMDQuaternion();
       hmdEuler.setFromQuaternion(hmdQuaternion, 'YXZ');
-      if (isMobile) {
-        // In mobile we allow camera rotation with touch events and sensors
+
+      if (el.sceneEl.isMobile) {
+        // In mobile, allow camera rotation with touch events and sensors.
         rotation = {
           x: radToDeg(hmdEuler.x) + radToDeg(pitchObject.rotation.x),
           y: radToDeg(hmdEuler.y) + radToDeg(yawObject.rotation.y),
           z: radToDeg(hmdEuler.z)
         };
       } else if (!sceneEl.is('vr-mode') || isNullVector(hmdEuler) || !this.data.hmdEnabled) {
-        currentRotation = this.el.getComputedAttribute('rotation');
+        // Mouse-drag only if HMD is not active (disabled, no incoming sensor data).
+        currentRotation = el.getComputedAttribute('rotation');
         deltaRotation = this.calculateDeltaRotation();
-        // Mouse look only if HMD disabled or no info coming from the sensors
         rotation = {
           x: currentRotation.x + deltaRotation.x,
           y: currentRotation.y + deltaRotation.y,
           z: currentRotation.z
         };
       } else {
-        // Mouse rotation ignored with an active headset.
-        // The user head rotation takes priority
+        // HMD. Mouse rotation ignored with active headset, user head rotation takes priority.
         rotation = {
           x: radToDeg(hmdEuler.x),
           y: radToDeg(hmdEuler.y),
           z: radToDeg(hmdEuler.z)
         };
       }
-      this.el.setAttribute('rotation', rotation);
+
+      el.setAttribute('rotation', rotation);
     };
   })(),
 
+  /**
+   * Calculate change in rotation by subtracting previous rotation from current rotation.
+   *
+   * @member previousRotationX
+   * @member previousRotationY
+   */
   calculateDeltaRotation: (function () {
-    var previousRotationX;
-    var previousRotationY;
+    var previousRotationX = 0;
+    var previousRotationY = 0;
     return function () {
       var currentRotationX = radToDeg(this.pitchObject.rotation.x);
       var currentRotationY = radToDeg(this.yawObject.rotation.y);
-      var deltaRotation;
-      previousRotationX = previousRotationX || currentRotationX;
-      previousRotationY = previousRotationY || currentRotationY;
```

Comment:
Actually, `calculateDeltaRotation` is not used for headsets. It is only used for mouse drag. I changed it to `0` because I caught in the unit tests that because it ended up doing:

```js
var deltaRotation = {
  x: currentRotationX - (previousRotationX || currentRotationX),
  y: currentRotationY - (previousRotationY || currentRotationY)
};
```

On the first tick then since `previousRotation` was undefined, we ended up:

```js
var deltaRotation = {
  x: currentRotationX - currentRotationX,
  y: currentRotationY - currentRotationY
};
```

And achieving no rotation